connector-sdk/jackson-datatype-feel/src/test/java/io/camunda/connector/feel/jackson/FeelFunctionDeserializerTest.java

```diff
@@ -195,4 +213,6 @@ private record TargetTypeInteger(Function<InputContextInteger, Integer> function
   private record TargetTypeList(Function<InputContextInteger, List<Long>> function) {}
 
   private record TargetTypeMap(Function<InputContextInteger, Map<String, Long>> function) {}
+
+  private record TargetTypeFoldedMap(Function<InputContextInteger, Map<String, Object>> function) {}
```

Comment:
nit:

The original issue actually involved an `Object` return type that contained a map as one of its fields, right? I think it might be more representative to test the similar use case here:

```
private record FoldedMapRecord(Map<String, Object> folded) {}

private record TargetTypeFoldedMap(Function<InputContextInteger, FoldedMapRecord> function) {}
```

The reason this caught my eye, I think the current test case could have been handled by the previous implementation as well (by [this code](https://github.com/camunda/connectors/blob/ab6a44e56660c39ce7c1bdd59f32e0df85345ed3/connector-sdk/feel-wrapper/src/main/java/io/camunda/connector/feel/FeelEngineWrapper.java#L101), as the root type is already a map). But I might be wrong here.

Reply:
Spent some time trying to rewrite and didn't manage to make it working 😕 
Let me merge it as-is now, and I'll follow up on this in [this task](https://github.com/camunda/connectors/issues/1354).